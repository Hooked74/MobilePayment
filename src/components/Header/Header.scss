@import "../../style/utils/variables";
@import "../../style/utils/functions";
@import "../../style/utils/mixins";

.container {
  display: flex;

  width: 100%;

  background-color: white !important;
  box-shadow: 0 2px 2px -2px #cfcfcf;

  align-items: center;
  justify-content: space-between;
}

.logo {
  font-size: 18px;

  cursor: pointer;
}

@include media("<=phone") {
  .logo {
    font-size: 16px;
  }

  .welcome {
    font-size: 12px;
  }
}

@include media("<=340px") {
  .logo {
    font-size: 14px;
  }

  .welcome {
    font-size: 11px;
  }
}
