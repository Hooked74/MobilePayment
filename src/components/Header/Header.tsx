import { Layout } from "antd";
import { NormalizedCacheObject } from "apollo-cache-inmemory";
import ApolloClient from "apollo-client";
import cookie from "cookie";
import Link from "next/link";
import { MouseEvent, PureComponent } from "react";
import * as React from "react";
import { redirect } from "../../utils";
import style from "./Header.scss";

interface IProps {
  name: string;
  client: ApolloClient<NormalizedCacheObject>;
}

export default class Header extends PureComponent<IProps> {
  public render() {
    return (
      <Layout.Header className={style.container}>
        <Link href="/" prefetch={true}>
          <div className={style.logo}>Mobile Payment</div>
        </Link>
        <div className={style.welcome}>
          <span>Hello {this.props.name}</span>,{" "}
          <a href="#" onClick={this.logout}>
            log out
          </a>
        </div>
      </Layout.Header>
    );
  }

  private logout = (e: MouseEvent) => {
    e.preventDefault();

    document.cookie = cookie.serialize("token", "", {
      maxAge: -1 // Expire the cookie immediately
    });

    // Force a reload of all the current queries now that the user is
    // logged in, so we don't accidentally leave any state around.
    this.props.client.cache.reset().then(() => {
      // Redirect to a more useful page when signed out
      redirect("/login");
    });
  };
}
