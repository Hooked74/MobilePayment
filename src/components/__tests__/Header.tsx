import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
import ApolloClient from "apollo-client";
import { createHttpLink } from "apollo-link-http";
import { shallow } from "enzyme";
import * as React from "react";
import * as fetch from "unfetch";
import Header from "../Header/Header";

let apolloClient: ApolloClient<NormalizedCacheObject>;
describe("Header", () => {
  beforeEach(() => {
    apolloClient = new ApolloClient({
      link: createHttpLink({ uri: "/graphql", fetch }),
      cache: new InMemoryCache().restore({})
    });
  });
  test("should render a Header component", () => {
    const wrapper = shallow(<Header name="mock" client={apolloClient} />);
    expect(wrapper).toMatchSnapshot();
  });
});
