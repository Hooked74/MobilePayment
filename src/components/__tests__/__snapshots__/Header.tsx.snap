// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a Header component 1`] = `
<Adapter>
  <Link
    href="/"
    prefetch={true}
  >
    <div>
      Mobile Payment
    </div>
  </Link>
  <div>
    <span>
      Hello 
      mock
    </span>
    ,
     
    <a
      href="#"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</Adapter>
`;
